Add documentation for instance display name feature (#2217)
nginx/documentation: content/nginx-one-console/agent/configure-instances/configure-instance-display-name.md

---
title: Assign a display name to an instance
toc: true
weight: 500
f5-content-type: how-to
f5-docs: DOCS-000
f5-product: NGINX Agent
---

## Overview

> By default, NGINX One Console identifies each NGINX instance by its hostname. You can assign a friendlier, custom name to an instance by setting the reserved `display-name` NGINX Agent label. Once NGINX Agent reports the label, NGINX One Console shows the display name alongside the hostname, and you can use it to filter and sort your instances.

Reviewer: ```suggestion
By default, NGINX One Console identifies each NGINX instance by its hostname. To assign a display name to an instance, set the reserved `display-name` NGINX Agent label. After NGINX Agent reports the label, NGINX One Console shows the display name alongside the hostname. You can then use the display name to filter and sort instances.
```

## Before you begin

Before you start, make sure that you have:

- [NGINX Agent installed]({{< ref "/nginx-one-console/agent/install-upgrade/" >}}) and connected to NGINX One Console.
- Access to the NGINX Agent configuration file, CLI, or container environment.

## Set a display name

Set the `display-name` label using any of the [NGINX Agent label]({{< ref "/nginx-one-console/agent/configure-instances/configuration-overview.md" >}}) configuration sources.

### Configuration file

Add the label to the `labels` section of `/etc/nginx-agent/nginx-agent.conf`:

```yaml
labels:
  display-name: prod-usw2-edge-03
```

### CLI parameters

```shell
nginx-agent --labels=display-name=prod-usw2-edge-03
```

### Environment variables

```shell
export NGINX_AGENT_LABELS="display-name=prod-usw2-edge-03"
```

After you save your changes, restart NGINX Agent to apply them:

```shell
sudo systemctl restart nginx-agent
```

{{< call-out class="note" >}}

> The display name can be up to 256 characters long. A value longer than this is rejected, and the instance keeps registering without a display name.

Reviewer: ```suggestion
Display names must be 256 characters or fewer. If a value exceeds this limit, the instance registers without a display name.
```

---
title: Assign a display name to an instance
toc: true
weight: 500
f5-content-type: how-to
f5-docs: DOCS-000

> f5-product: NGINX Agent

Reviewer: ```suggestion
  title: Assign a display name to an instance
  description: "Set a custom display name for an NGINX instance in NGINX One Console using the NGINX Agent display-name
  label."
  toc: true
  weight: 500
  f5-content-type: how-to
  f5-product: NGINX Agent
  f5-keywords: "display name, instance name, nginx agent label, nginx one console"
  f5-summary: >
    Assign a custom display name to an NGINX instance by setting the reserved
    display-name label in NGINX Agent. Use it to identify and filter instances
    in NGINX One Console instead of relying on hostname alone.
  f5-audience: operator
```

Before you start, make sure that you have:

- [NGINX Agent installed]({{< ref "/nginx-one-console/agent/install-upgrade/" >}}) and connected to NGINX One Console.
- Access to the NGINX Agent configuration file, CLI, or container environment.

## Set a display name

Set the `display-name` label using any of the [NGINX Agent label]({{< ref "/nginx-one-console/agent/configure-instances/configuration-overview.md" >}}) configuration sources.

### Configuration file

Add the label to the `labels` section of `/etc/nginx-agent/nginx-agent.conf`:

```yaml
labels:
  display-name: prod-usw2-edge-03
```

### CLI parameters

```shell
nginx-agent --labels=display-name=prod-usw2-edge-03
```

### Environment variables

```shell
export NGINX_AGENT_LABELS="display-name=prod-usw2-edge-03"
```

After you save your changes, restart NGINX Agent to apply them:

```shell
sudo systemctl restart nginx-agent
```

{{< call-out class="note" >}}
Display names must be 256 characters or fewer. If a value exceeds this limit, the instance registers without a display name.
{{< /call-out >}}

{{< call-out class="note" >}}
Display names cannot include Unicode characters like emoji or other multi-byte characters.

> {{< /call-out >}}

Reviewer: ```suggestion
{{< call-out class="important" >}}
Display names must be 256 characters or fewer. If a value exceeds this limit, the instance registers without a display name.

Display names cannot include Unicode characters like emoji or other multi-byte characters.
{{< /call-out >}}
```
